pywr/_recorders.pyx

```diff
@@ -1,20 +1,44 @@
 import numpy as np
 cimport numpy as np
-
 from pywr._core cimport Timestep
-
 import pandas as pd
+from past.builtins import basestring
 
 recorder_registry = set()
 
+cdef enum AggFuncs:
```

Comment:
I don't think the enums need to be cpdef. They're just used so the compiler can optimise the switch/case statement. The end-user is expected to use strings (e.g. "sum").